File: 승준/20230623/virus_2606.js
const readline = require("readline");
let v, w;
const graph = [];
const input = [];

readline
  .createInterface(process.stdin, process.stdout)
  .on("line", (line) => {
    input.push(line);
  })
  .on("close", () => {
    v = parseInt(input[0]);
    w = parseInt(input[1]);

    const visited = new Array(v + 1).fill(false);

    for (let j = 0; j < v + 1; j++) {
      graph.push(new Array());
    }

    for (let i = 2; i < input.length; i++) {
      const [a, b] = input[i].split(" ").map(Number);
      graph[a].push(b);
      graph[b].push(a);
    }

    const queue = [];
    queue.push(1);
    visited[1] = true;
    let count = 0;

    while (queue.length > 0) {
      const node = queue.shift();

      for (const next of graph[node]) {
        if (!visited[next]) {
          count++;
          queue.push(next);
          visited[next] = true;
        }
      }
    }

    console.log(count);
    process.exit();
  });

File: 승준/20230623/virus_2606.py
from collections import deque

v = int(input())
w = int(input())
graph = [[] for _ in range(v + 1)]
visited = [False] * (v + 1)

for _ in range(w):
    a, b = map(int, input().split())
    graph[a].append(b)
    graph[b].append(a)

start = 1
queue = deque()
queue.append(start)
visited[start] = True
count = 0

while queue:
    node = queue.popleft()

    for next_node in graph[node]:
        if not visited[next_node]:
            count += 1
            queue.append(next_node)
            visited[next_node] = True

print(count)